src/loggers/environment.logger.ts

```diff
@@ -0,0 +1,51 @@
+import { Environment } from '../environment'
+import { Format } from '../format'
+import { JsonFormat } from '../formats/json.format'
+import { LoggerBase } from '../logger-base'
+import { Level } from '../level'
+import { Vault } from '../vault'
+import { ColoredPlainTextFormat } from '../formats/colored-plain-text.format'
+
+export class EnvironmentLogger extends LoggerBase {
+
+  private readonly environment: Environment
+  private level: Level
+
+  constructor(vaults: Vault[], environment: Environment, level: Level) {
+    super(vaults)
+
+    this.environment = environment
+    this.level = level
+
+    this.applyEnvironment()
+  }
+
+  private applyEnvironment(): void {
+    this.setEnvironment(this.environment)
+    this.setLevel(this.level)
+  }
+
+  public setLevel = (level: Level): void => {
+    this.level = level
+    super.setLevel(level)
+  }
```

Comment:
Why is this created as a variable containing a function, instead of simply a function?

```suggestion
  public setLevel(level: Level): void {
    this.level = level
    super.setLevel(level)
  }
```

Reply:
It is a remnant of the code prior to the `MethodBinder` to ensure that the `this` is bound correctly when given to a higher order function. Will be corrected.